test/jdk/java/awt/print/PrinterJob/PrintLatinCJKTest.java

```diff
@@ -31,59 +31,61 @@
  * @run main/manual PrintLatinCJKTest
  */
 
-import java.awt.BorderLayout;
 import java.awt.Font;
 import java.awt.Graphics;
 import java.awt.print.PageFormat;
 import java.awt.print.Printable;
 import java.awt.print.PrinterException;
 import java.awt.print.PrinterJob;
 import java.lang.reflect.InvocationTargetException;
-import javax.swing.JButton;
-import javax.swing.JFrame;
 
-import static javax.swing.SwingUtilities.invokeAndWait;
+import javax.swing.BorderFactory;
+import javax.swing.Box;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JOptionPane;
 
 public class PrintLatinCJKTest implements Printable {
 
-    private static PrintLatinCJKTest testInstance = new PrintLatinCJKTest();
-    private static JFrame frame;
-    private static final String info = """
+    private static final String TEXT = "\u4e00\u4e01\u4e02\u4e03\u4e04English";
+
+    private static final String INFO = """
             You need a printer for this test. If you have none, let
```

Comment:
Can we use `PrinterJob.lookupPrintServices().length` to check for printer availability?

Reply:
Yes, we can. There's `@key printer` which implies the test is to be run on a system where at least one printer is available.

What do you suggest actually?

Reply:
Yeah, makes sense.